packages/legend-extension-external-store-service/src/components/ESService_ServiceStoreEmbeddedData.tsx

```diff
@@ -912,44 +912,36 @@ export const ServiceStoreEmbeddedDataEditor = observer(
                 </button>
               </div>
             </div>
-            <ContextMenu
-              className="service-store-embedded-data-editor__content"
-              disabled={isReadOnly}
-              content={
-                !isReadOnly &&
-                currentServiceStubMapping !== undefined && (
+            {serviceStoreEmbeddedDataState.embeddedData.serviceStubMappings.map(
+              (serviceStubMapping: ServiceStubMapping, index: number) => (
+                <ContextMenu
+                  key={serviceStubMapping._UUID}
+                  className={clsx('service-store-embedded-data-editor__item', {
+                    'service-store-embedded-data-editor__item--active':
+                      currentServiceStubMapping === serviceStubMapping,
+                  })}
+                  disabled={isReadOnly}
+                  content={
+                    !isReadOnly && (
```

Comment:
I think the `disabled` flag is enough to safeguard this, also, let's use `Menu` component so you don't need any `CSS` here.